module/qr_code.py

```diff
@@ -1,5 +1,162 @@
+        return "{}, {}".format(repr(self.x), repr(self.y))
+
+
+class Line:
+    """ This class is used to represent two points of a line. """
+
+    def __init__(self, p1=Coordinate(), p2=Coordinate()):
+        """ The constructor.
+        :param p1: First coordinate of the line. Default = empty coordinate. 
+        :param p2: Second coordinate of the line. Default = empty coordinate.
+        :return: Nothing.
+        """
+        self.p1 = p1
+        self.p2 = p2
+
+    def __repr__(self):
+        """ Represents the line
+        :return: Returns a string with the two coordinates.
+        """
+        return "{}, {}".format(repr(self.p1), repr(self.p2))
+
+
+class QrPolygon:
+    """ This class extends the information one gets from the pyzbar polygon."""
+
+    def __init__(self, polygon):
+        """ The constructor.
+
+        It creates a lot of lines and calls the calculate_polygon_properties function to calculate more information.
+        :param polygon: Either a pyzbar polygon from a qr code, or a list with four coordinates.
+        :return: Nothing.
+        """
+        self.top_left_point = polygon[0]
+        self.bottom_left_point = polygon[1]
+        self.bottom_right_point = polygon[2]
+        self.top_right_point = polygon[3]
+
+        self.diagonal_tl_br_line = Line(
+            self.top_left_point, self.bottom_right_point)
+        self.diagonal_tr_bl_line = Line(
+            self.top_right_point, self.bottom_left_point)
+
+        self.top_line = Line(self.top_left_point, self.top_right_point)
+        self.bottom_line = Line(self.bottom_left_point,
+                                self.bottom_right_point)
+        self.left_line = Line(self.top_left_point, self.bottom_left_point)
+        self.right_line = Line(self.top_right_point, self.bottom_right_point)
+
+        self.center_point = Coordinate()
+        self.upper_middle_point = Coordinate()
+        self.lower_middle_point = Coordinate()
+        self.left_middle_point = Coordinate()
+        self.right_middle_point = Coordinate()
+        self.middle_height_line = Line()
+        self.middle_width_line = Line()
+        self.middle_height = 0
+        self.middle_width = 0
+
+        self.horizontal_vanish_point = Coordinate()
+        self.vertical_vanish_point = Coordinate()
+
+        self.horizontal_center_line = Line()
+        self.vertical_center_line = Line()
+
+        self.calculate_polygon_properties()
+
+    def line_intersection(self, line_1, line_2):
+        """ Calculates the intersection coordinate of two lines.
+        :param line_1: A line.
+        :param line_2: A line.
+        :return: Coordinate of the intersection point.
+        :return: Nothing if no point is found.
+        """
+        x_diff = Coordinate(line_1.p1.x - line_1.p2.x,
+                            line_2.p1.x - line_2.p2.x)
+        y_diff = Coordinate(line_1.p1.y - line_1.p2.y,
+                            line_2.p1.y - line_2.p2.y)
+
+        def det(a, b):
```

Comment:
Please add the reasing for nesting this function instead of making it a private-hinted class function.

Reply:
Added reason on lines 113 through 116 in commit https://github.com/R2D2-2019/vision/pull/11/commits/2d4a20c0fd93867a7d668319a7c45ef5a00d22f7